stdlib/_socket.pyi

```diff
@@ -0,0 +1,619 @@
+    TIPC_SRC_DROPPABLE: int
+    TIPC_SUBSCR_TIMEOUT: int
+    TIPC_SUB_CANCEL: int
+    TIPC_SUB_PORTS: int
+    TIPC_SUB_SERVICE: int
+    TIPC_TOP_SRV: int
+    TIPC_WAIT_FOREVER: int
+    TIPC_WITHDRAWN: int
+    TIPC_ZONE_SCOPE: int
+
+if sys.platform == "linux":
+    AF_ALG: int
+    SOL_ALG: int
+    ALG_OP_DECRYPT: int
+    ALG_OP_ENCRYPT: int
+    ALG_OP_SIGN: int
+    ALG_OP_VERIFY: int
+    ALG_SET_AEAD_ASSOCLEN: int
+    ALG_SET_AEAD_AUTHSIZE: int
+    ALG_SET_IV: int
+    ALG_SET_KEY: int
+    ALG_SET_OP: int
+    ALG_SET_PUBKEY: int
+
+if sys.platform == "linux" and sys.version_info >= (3, 7):
+    AF_VSOCK: int
+    IOCTL_VM_SOCKETS_GET_LOCAL_CID: int
+    VMADDR_CID_ANY: int
+    VMADDR_CID_HOST: int
+    VMADDR_PORT_ANY: int
+    SO_VM_SOCKETS_BUFFER_MAX_SIZE: int
+    SO_VM_SOCKETS_BUFFER_SIZE: int
+    SO_VM_SOCKETS_BUFFER_MIN_SIZE: int
+    VM_SOCKETS_INVALID_VERSION: int
+
+AF_LINK: int  # Availability: BSD, macOS
+
+# BDADDR_* and HCI_* listed with other bluetooth constants below
+
+SO_DOMAIN: int
+SO_PASSSEC: int
+SO_PEERSEC: int
+SO_PROTOCOL: int
+TCP_CONGESTION: int
+TCP_USER_TIMEOUT: int
+
+if sys.platform == "linux" and sys.version_info >= (3, 8):
+    AF_QIPCRTR: int
+
+# Semi-documented constants
+# (Listed under "Socket families" in the docs, but not "Constants")
+
+if sys.platform == "linux":
+    # Netlink is defined by Linux
+    AF_NETLINK: int
+    NETLINK_ARPD: int
+    NETLINK_CRYPTO: int
+    NETLINK_DNRTMSG: int
+    NETLINK_FIREWALL: int
+    NETLINK_IP6_FW: int
+    NETLINK_NFLOG: int
+    NETLINK_ROUTE6: int
+    NETLINK_ROUTE: int
+    NETLINK_SKIP: int
+    NETLINK_TAPBASE: int
+    NETLINK_TCPDIAG: int
+    NETLINK_USERSOCK: int
+    NETLINK_W1: int
+    NETLINK_XFRM: int
+
+if sys.platform != "win32" and sys.platform != "darwin":
+    # Linux and some BSD support is explicit in the docs
+    # Windows and macOS do not support in practice
+    AF_BLUETOOTH: int
+    BTPROTO_HCI: int
+    BTPROTO_L2CAP: int
+    BTPROTO_RFCOMM: int
+    BTPROTO_SCO: int  # not in FreeBSD
+
+    BDADDR_ANY: str
+    BDADDR_LOCAL: str
+
+    HCI_FILTER: int  # not in NetBSD or DragonFlyBSD
+    # not in FreeBSD, NetBSD, or DragonFlyBSD
+    HCI_TIME_STAMP: int
+    HCI_DATA_DIR: int
+
+if sys.platform == "darwin":
+    # PF_SYSTEM is defined by macOS
+    PF_SYSTEM: int
+    SYSPROTO_CONTROL: int
+
+# ----- Exceptions -----
+
+error = OSError
+
+class herror(error):
+    def __init__(self, herror: int = ..., string: str = ...) -> None: ...
+
+class gaierror(error):
+    def __init__(self, error: int = ..., string: str = ...) -> None: ...
+
+class timeout(error):
+    def __init__(self, error: int = ..., string: str = ...) -> None: ...
+
+# ----- Classes -----
+
+class socket:
+    family: int
+    type: int
+    proto: int
+    def __init__(self, family: int = ..., type: int = ..., proto: int = ..., fileno: Optional[int] = ...) -> None: ...
+    def bind(self, address: Union[_Address, bytes]) -> None: ...
```

Comment:
Missing the close method
```suggestion
    def bind(self, address: Union[_Address, bytes]) -> None: ...
    def close(self) -> None: ...
```

Reply:
Also, bind only takes positional args. The same is true for the other METH_O methods: bind, connect, connect_ex, setblocking, settimeout, shutdown.

Reply:
Also a few functions and everything marked with `METH_VARARGS`. Changing.